generation/WinSDK/emitter.settings.rsp

```diff
@@ -1824,29 +1831,30 @@ GetFileVersionInfoSizeA=[CanReturnMultipleSuccessValues]
 GetFileVersionInfoSizeW=[CanReturnMultipleSuccessValues]
 CertCreateCertificateContext=[CanReturnMultipleSuccessValues]
 DeviceIoControl::lpOverlapped=[In][Out][Optional][Retained]
-DXGI_ADAPTER_DESC1::Flags=[AssociatedEnum("DXGI_ADAPTER_FLAG")]
-DXGI_ADAPTER_DESC2::Flags=[AssociatedEnum("DXGI_ADAPTER_FLAG")]
-DXGI_DECODE_SWAP_CHAIN_DESC::Flags=[AssociatedEnum("DXGI_SWAP_CHAIN_FLAG")]
-DXGI_SWAP_CHAIN_DESC::Flags=[AssociatedEnum("DXGI_SWAP_CHAIN_FLAG")]
-DXGI_SWAP_CHAIN_DESC1::Flags=[AssociatedEnum("DXGI_SWAP_CHAIN_FLAG")]
-IDXGIDevice4::OfferResources1::Flags=[AssociatedEnum("DXGI_OFFER_RESOURCE_FLAGS")]
-IDXGIOutput3::CheckOverlaySupport::pFlags=[AssociatedEnum("DXGI_OVERLAY_SUPPORT_FLAG")]
-IDXGIOutput4::CheckOverlayColorSpaceSupport::pFlags=[AssociatedEnum("DXGI_OVERLAY_COLOR_SPACE_SUPPORT_FLAG")]
-IDXGIOutput6::CheckHardwareCompositionSupport::pFlags=[AssociatedEnum("DXGI_HARDWARE_COMPOSITION_SUPPORT_FLAGS")]
-IDXGISwapChain::ResizeBuffers::SwapChainFlags=[AssociatedEnum("DXGI_SWAP_CHAIN_FLAG")]
-IDXGISwapChain3::CheckColorSpaceSupport::pColorSpaceSupport=[AssociatedEnum("DXGI_SWAP_CHAIN_COLOR_SPACE_SUPPORT_FLAG")]
-IDXGISwapChain3::ResizeBuffers1::SwapChainFlags=[AssociatedEnum("DXGI_SWAP_CHAIN_FLAG")]
+DXGI_ADAPTER_DESC1::Flags=DXGI_ADAPTER_FLAG
+DXGI_ADAPTER_DESC2::Flags=DXGI_ADAPTER_FLAG
+DXGI_DECODE_SWAP_CHAIN_DESC::Flags=DXGI_SWAP_CHAIN_FLAG
+DXGI_SWAP_CHAIN_DESC::Flags=DXGI_SWAP_CHAIN_FLAG
+DXGI_SWAP_CHAIN_DESC1::Flags=DXGI_SWAP_CHAIN_FLAG
 IDXGIAdapter::GetDesc::pDesc=[RetVal]
 IDXGIAdapter1::GetDesc1::pDesc=[RetVal]
 IDXGIAdapter2::GetDesc2::pDesc=[RetVal]
 IDXGIAdapter4::GetDesc3::pDesc=[RetVal]
+IDXGIDevice4::OfferResources1::Flags=[AssociatedEnum("DXGI_OFFER_RESOURCE_FLAGS")]
```

Comment:
What do folks think, should we proactively convert other `enum` types - that aren't used in structs _just yet_ - to `uint` as well to use them as a direct type instead of via `AssociatedEnum`?

@kennykerr we've seen it before in https://github.com/microsoft/windows-rs/issues/3111 that `AssociatedEnum` like this cause `impl` wrappers to borrow instead of pass by (cheaply copyable) value. Is that something you have on your radar to fix in `windows-rs`?

Reply:
> What do folks think, should we proactively convert other `enum` types - that aren't used in structs _just yet_ - to `uint` as well to use them as a direct type instead of via `AssociatedEnum`?

I've just pushed this and regenerated `windows-rs` at https://github.com/MarijnS95/windows-rs/commit/fbd504aabb4dcac1d6f8f878c85ff254a66d4709. Notable takeaways:

- `AssociatedEnum` doesn't seem to work for to `[out]` pointers, for example `CheckHardwareCompositionSupport()` was still returning `u32` but now finally returns `DXGI_HARDWARE_COMPOSITION_SUPPORT_FLAGS`;
- All `flags: &` borrows on `_Impl` traits are gone, and now passed by values;
- (less relevant: less internal conversions; raw function signatures now use the underlying flags type more early).